src/backend/joanie/core/models/courses.py

```diff
@@ -160,6 +162,64 @@ def __str__(self):
         )
 
 
+class OrganizationAccess(BaseModel):
+    """Link table between organizations and users"""
+
+    OWNER = "owner"
+    ADMIN = "administrator"
+    MEMBER = "member"
+
+    ROLE_CHOICES = (
+        (OWNER, _("owner")),
+        (ADMIN, _("administrator")),
+        (MEMBER, _("member")),
+    )
+
+    organization = models.ForeignKey(
+        Organization,
+        on_delete=models.CASCADE,
+        related_name="accesses",
+    )
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="accesses")
+    role = models.CharField(max_length=20, choices=ROLE_CHOICES, default=MEMBER)
+
+    class Meta:
+        db_table = "joanie_organization_access"
+        verbose_name = _("Organization access")
+        verbose_name_plural = _("Organization accesses")
+        unique_together = ("organization", "user")
+
+    def __str__(self):
+        role = capfirst(self.get_role_display())
+        return (
+            f"{role:s} role for {self.user.username:s} on {self.organization.title:s}"
+        )
+
+    def save(self, *args, **kwargs):
+        """Make sure we keep at least one owner for the organization."""
+        if self.pk and self.role != self.OWNER:
+            accesses = self._meta.model.objects.filter(
+                organization=self.organization, role=self.OWNER
+            ).only("pk")
+            if len(accesses) == 1 and accesses[0].pk == self.pk:
+                raise PermissionDenied(
+                    "An organization should keep at least one owner."
+                )
+        return super().save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        """Disallow deleting the last owner."""
+        if (
+            self.role == self.OWNER
+            and self._meta.model.objects.filter(
+                organization=self.organization, role=self.OWNER
+            ).count()
+            == 1
+        ):
+            raise PermissionDenied("An organization should keep at least one owner.")
```

Comment:
I feel we could factorize a little bit checks executed in delete and save. WDYT ?

Reply:
Yes indeed, I read too quickly the if statement.